src/container.cpp

```diff
@@ -622,14 +620,14 @@ void Container::removeThing(Thing* thing, uint32_t count)
 		}
 
 		item->setParent(nullptr);
-		itemlist.erase(itemlist.begin() + index);
+		itemList.erase(itemList.begin() + index);
 	}
 }
 
-int32_t Container::getThingIndex(const Thing* thing) const
+int32_t Container::getThingIndex(std::shared_ptr<const Thing> thing) const
 {
 	int32_t index = 0;
-	for (Item* item : itemlist) {
+	for (auto item : itemList) {
```

Comment:
Could this be `const auto`
```suggestion
	for (const auto& item : itemList) {
```